### DISCUSSION TITLE
How to disable built-in activities registration?

### DISCUSSION THREAD

**Discussion by defunky** [elsa-workflows/elsa-core#5183] [Q&A]
In V2, there was a method named `NoCoreActivities` that’s not present in V3. I’ve looked into the source code and found a property called `DisableAutomaticActivityRegistration `within `ElsaFeature`. Inserting `elsa.Configure<ElsaFeature>(x => x.DisableAutomaticActivityRegistration = true)` within the `services.AddElsa()` didn’t seem to have any effect. Adding it as a chained method after `.AddElsa()` extension method was also unsuccessful.

I even tried to create a separate extension method similar to AddElsa, placing `module.Configure<ElsaFeature>(x => x.DisableAutomaticActivityRegistration = true)`; before the `Apply `method. This approach seemed to trigger a reaction, as it resulted in an `exception: System.Exception: Activity descriptor not found`. What would be the correct approach to utilize this feature?

**Comment by glime-ai** [comment]
To disable automatic activity registration in Elsa 3, you should configure the `ElsaOptions` during the service registration process. Based on the context provided, it seems that the `DisableAutomaticActivityRegistration` property is part of the `ElsaFeature` options. You should be able to configure this during the `AddElsa` call by using the `Configure` extension method on the `IServiceCollection`.

Here's an example of how you might configure this:

```csharp
services.AddElsa(elsa =>
{
    // Other Elsa setup...
})
.Configure<ElsaFeature>(feature =>
{
    feature.DisableAutomaticActivityRegistration = true;
});
```

If this approach does not work, it's possible that the feature might not be fully implemented or there could be a bug. Since you mentioned that you've already tried a similar approach without success, it might be worth checking the Elsa GitHub issues or discussions for any known issues or updates regarding this feature. If you don't find a solution, consider opening a new issue or discussion on the Elsa GitHub repository to get help from the maintainers or the community.

Remember to check the official documentation and the source code for any updates or changes to the API that might affect how this feature should be used.

**Comment by defunky** [comment]
@sfmskywalker I think you implemented this property, can you help here

**Comment by cr0fters** [comment]
I'm struggling with the same issue, have you got anywhere with it? It feels like it's a bug and doesn't work as it should

**Comment by defunky** [comment]
nope, you basically have a 1 in million chance that the the only maintainer replies

**Reply by sfmskywalker** [reply]
That’s a better chance than being struck by lightning ;) I’ll look into it soon.

**Comment by cr0fters** [comment]
hey @defunky - I've created an extension class that removes all the default activities, you just need to call it via `elsa.RemoveDefaultActivities();`

Here you go:

```
using Elsa.Extensions;
using Elsa.Features.Services;
using Elsa.Http;
using Elsa.Scheduling.Activities;
using Elsa.Workflows.Activities;
using Elsa.Workflows.Activities.Flowchart.Activities;
using Elsa.Workflows.Management.Activities.SetOutput;
using Elsa.Workflows.Runtime.Activities;
using Timer = Elsa.Scheduling.Activities.Timer;

namespace ElsaServer.Extensions;

public static class ElsaExtensions
{
    public static void RemoveDefaultActivities(this IModule elsa)
    {
        elsa.RemoveActivity<If>();
        elsa.RemoveActivity<FlowDecision>();
        elsa.RemoveActivity<FlowFork>();
        elsa.RemoveActivity<FlowJoin>();
        elsa.RemoveActivity<FlowSwitch>();
        elsa.RemoveActivity<Switch>();
    
        elsa.RemoveActivity<BulkDispatchWorkflows>();
        elsa.RemoveActivity<Complete>();
        elsa.RemoveActivity<DispatchWorkflow>();
        elsa.RemoveActivity<SetOutput>();
    
        elsa.RemoveActivity<ReadLine>();
        elsa.RemoveActivity<WriteLine>();
    
        elsa.RemoveActivity<FlowNode>();
        elsa.RemoveActivity<End>();
        elsa.RemoveActivity<Start>();
    
        elsa.RemoveActivity<HttpEndpoint>();
        elsa.RemoveActivity<WriteFileHttpResponse>();
        elsa.RemoveActivity<SendHttpRequest>();
        elsa.RemoveActivity<FlowSendHttpRequest>();
        elsa.RemoveActivity<WriteHttpResponse>();
    
        elsa.RemoveActivity<Break>();
        elsa.RemoveActivity<For>();
        elsa.RemoveActivity<ForEach>();
        elsa.RemoveActivity<ParallelForEach>();
        elsa.RemoveActivity<While>();
    
        elsa.RemoveActivity<Correlate>();
        elsa.RemoveActivity<Event>();
        elsa.RemoveActivity<Fault>();
        elsa.RemoveActivity<Finish>();
        elsa.RemoveActivity<PublishEvent>();
        elsa.RemoveActivity<RunTask>();
        elsa.RemoveActivity<SetName>();
        elsa.RemoveActivity<SetVariable>();
    
        elsa.RemoveActivity<Cron>();
        elsa.RemoveActivity<Delay>();
        elsa.RemoveActivity<StartAt>();
        elsa.RemoveActivity<Timer>();
    }
}
```

**Reply by defunky** [reply]
Cheers mate
